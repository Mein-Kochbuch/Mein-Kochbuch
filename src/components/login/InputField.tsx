import React from 'react';
import styled from 'styled-components/native';
import {KeyboardTypeOptions, Text} from 'react-native';
// @ts-ignore
import Colors from 'react-native/Libraries/NewAppScreen/components/Colors';
import {Appearance} from 'react-native';

interface InputFieldProps {
  title: string;
  errorText?: string;
  onChangeText: (value: string) => void;
  placeholder: string;
  value: string;
  keyboardType?: KeyboardTypeOptions;
  secureTextEntry?: boolean;
}

export default function InputField({
  title,
  errorText,
  ...props
}: InputFieldProps) {
  const isDarkMode = Appearance.getColorScheme() === 'dark';

  return (
    <StyledView>
      <Text style={errorText ? {color: '#ff0000'} : {}}>
        {errorText ? errorText : title}
      </Text>
      <StyledTextInput isDarkMode={isDarkMode} {...props} />
    </StyledView>
  );
}

const StyledView = styled.View`
  margin: 12px;
`;

interface StyledTextInputProps {
  isDarkMode: boolean;
  onChangeText: (value: string) => void;
  placeholder: string;
  value: string;
  keyboardType?: string;
  secureTextEntry?: boolean;
}

const StyledTextInput = styled.TextInput<StyledTextInputProps>`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: 80%;
  min-width: 300px;
  border-style: solid;
  border-color: ${props => (props.isDarkMode ? Colors.lighter : Colors.darker)};
  border-radius: 10px;
  border-width: 2px;
`;
