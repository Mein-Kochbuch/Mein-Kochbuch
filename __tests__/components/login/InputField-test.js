import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {useColorScheme} from "react-native-appearance";
import InputField from "../../../src/components/login/InputField";

jest.mock("react-native-appearance", () => ({
    useColorScheme: jest.fn(),
}))

describe('InputField Test', () => {

    afterEach(() => {
        jest.clearAllMocks();
    });


    it('Snapshot light Test', () => {
        useColorScheme.mockReturnValueOnce('light');
        const onChangeText = jest.fn()

        const component = renderer.create(
            <InputField onChangeText={onChangeText} placeholder={"test-palceholder"} title={"test-title"} value={""}/>
        );

        expect(component.toJSON()).toMatchSnapshot()
    })

    it('Snapshot dark Test', () => {
        useColorScheme.mockReturnValueOnce('dark');
        const onChangeText = jest.fn()

        const component = renderer.create(
            <InputField onChangeText={onChangeText} placeholder={"test-palceholder"} title={"test-title"} value={""}/>
        );

        expect(component.toJSON()).toMatchSnapshot()
    })

    it('Snapshot with value', () => {
        useColorScheme.mockReturnValueOnce('dark');
        const onChangeText = jest.fn()

        const component = renderer.create(
            <InputField onChangeText={onChangeText} placeholder={"test-palceholder"} title={"test-title"}
                        value={"test-value"}/>
        );

        expect(component.toJSON()).toMatchSnapshot()
    })

    it('Snapshot with value, keyboard type email', () => {
        useColorScheme.mockReturnValueOnce('dark');
        const onChangeText = jest.fn()

        const component = renderer.create(
            <InputField onChangeText={onChangeText} placeholder={"test-palceholder"} title={"test-title"}
                        value={"test-value"} keyboardType={"email-address"}/>
        );

        expect(component.toJSON()).toMatchSnapshot()
    })

    it('Snapshot with value, secureTextEntry', () => {
        useColorScheme.mockReturnValueOnce('dark');
        const onChangeText = jest.fn()

        const component = renderer.create(
            <InputField onChangeText={onChangeText} placeholder={"test-palceholder"} title={"test-title"}
                        value={"test-value"} secureTextEntry={true}/>
        );

        expect(component.toJSON()).toMatchSnapshot()
    })

    it('Snapshot with value, ErrorTest', () => {
        useColorScheme.mockReturnValueOnce('dark');
        const onChangeText = jest.fn()

        const component = renderer.create(
            <InputField onChangeText={onChangeText} placeholder={"test-palceholder"} title={"test-title"}
                        value={"test-value"} errorText={"test-Error"}/>
        );

        expect(component.toJSON()).toMatchSnapshot()
    })
})
