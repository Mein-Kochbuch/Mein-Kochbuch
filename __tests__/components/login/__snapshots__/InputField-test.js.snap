// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Test Snapshot dark Test 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 12,
        "marginLeft": 12,
        "marginRight": 12,
        "marginTop": 12,
      },
    ]
  }
>
  <Text
    style={Object {}}
  >
    test-title
  </Text>
  <TextInput
    isDarkMode={true}
    onChangeText={[MockFunction]}
    placeholder="test-palceholder"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#F3F3F3",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "minWidth": 300,
          "width": "80%",
        },
      ]
    }
    value=""
  />
</View>
`;

exports[`InputField Test Snapshot light Test 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 12,
        "marginLeft": 12,
        "marginRight": 12,
        "marginTop": 12,
      },
    ]
  }
>
  <Text
    style={Object {}}
  >
    test-title
  </Text>
  <TextInput
    isDarkMode={false}
    onChangeText={[MockFunction]}
    placeholder="test-palceholder"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#222",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "minWidth": 300,
          "width": "80%",
        },
      ]
    }
    value=""
  />
</View>
`;

exports[`InputField Test Snapshot with value 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 12,
        "marginLeft": 12,
        "marginRight": 12,
        "marginTop": 12,
      },
    ]
  }
>
  <Text
    style={Object {}}
  >
    test-title
  </Text>
  <TextInput
    isDarkMode={true}
    onChangeText={[MockFunction]}
    placeholder="test-palceholder"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#F3F3F3",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "minWidth": 300,
          "width": "80%",
        },
      ]
    }
    value="test-value"
  />
</View>
`;

exports[`InputField Test Snapshot with value, ErrorTest 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 12,
        "marginLeft": 12,
        "marginRight": 12,
        "marginTop": 12,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#ff0000",
      }
    }
  >
    test-Error
  </Text>
  <TextInput
    isDarkMode={true}
    onChangeText={[MockFunction]}
    placeholder="test-palceholder"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#F3F3F3",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "minWidth": 300,
          "width": "80%",
        },
      ]
    }
    value="test-value"
  />
</View>
`;

exports[`InputField Test Snapshot with value, keyboard type email 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 12,
        "marginLeft": 12,
        "marginRight": 12,
        "marginTop": 12,
      },
    ]
  }
>
  <Text
    style={Object {}}
  >
    test-title
  </Text>
  <TextInput
    isDarkMode={true}
    keyboardType="email-address"
    onChangeText={[MockFunction]}
    placeholder="test-palceholder"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#F3F3F3",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "minWidth": 300,
          "width": "80%",
        },
      ]
    }
    value="test-value"
  />
</View>
`;

exports[`InputField Test Snapshot with value, secureTextEntry 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 12,
        "marginLeft": 12,
        "marginRight": 12,
        "marginTop": 12,
      },
    ]
  }
>
  <Text
    style={Object {}}
  >
    test-title
  </Text>
  <TextInput
    isDarkMode={true}
    onChangeText={[MockFunction]}
    placeholder="test-palceholder"
    secureTextEntry={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#F3F3F3",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "minWidth": 300,
          "width": "80%",
        },
      ]
    }
    value="test-value"
  />
</View>
`;
